### PR TITLE
[4.4] Fix ordering for files in the template view

### PR DESCRIPTION
**Pull request by HLeithner**
Pull Request for Issue #40605 .

### Summary of Changes
Order is now using the right values for displaying directory structure for templates in the backend.
This could happen on posix filesystems because the natural order is the file creation and not the alphabetical order.

### Testing Instructions
Check if your directories and files are sorted in the corrected order

### Actual result BEFORE applying this Pull Request

The order was only for directories correct

### Expected result AFTER applying this Pull Request

Sorting happens in the model and not longer in the view for directories and files.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### Additional
image

### CODE REVIEW

File: administrator/components/com_templates/src/Model/TemplateModel.php [hunk 1]
```diff
@@ -430,6 +430,24 @@ public function getDirectoryTree($dir)
...
+            if (\is_array($a)) {
+                if (\is_array($b)) {
+                    return key($a) <=> key($b);
+                }
+
+                return -1;
+            }
+
+            if (\is_array($b)) {
+
```
Comment (dgrammatiko) [inline administrator/components/com_templates/src/Model/TemplateModel.php]:
```suggestion
```

**Comment by dautrich** [comment]
@HLeithner 
The fix is for J4, isn't it? Can I apply the patch to J5.1 as well?

**Comment by brianteeman** [comment]
Is the image a before or after. If after then it is showing the folders in the tree not being in alphabetical order

**Comment by HLeithner** [comment]
I think so, I didn't see much changes since 4.4 in 5.1

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on f3a8732a925047651965d23533928c1afb5c4a13<br /><br />I successfully tested the PR on a J5.1.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43335">issues.joomla.org/tracker/joomla-cms/43335</a>.</sub>

**Comment by HLeithner** [comment]
> Is the image a before or after. If after then it is showing the folders in the tree not being in alphabetical order

correct php fooled me and the over complicated php code didn't make it better. I updated the code and moved from uasort to uksort. Now it works as expected.

Here the image from joomla 5.1:
image

**Comment by HLeithner** [comment]
> Now I am really confused. The original PR used uasort and did not work with cassiopeia. Yet @dautrich says it does work with the j51 template

it partly works with cassiopeia as you pointed out based on the image (media folder -> images shouldn't be on the last place, btw. this information in you comment would have saved me 20 minutes...). so his test is wrong.

**Comment by dautrich** [comment]
### Template Skylar

grafik

**Comment by brianteeman** [comment]
Please review https://github.com/joomla/joomla-cms/pull/25792/

**Comment by HLeithner** [comment]
> Please review #25792

done

**Comment by dautrich** [comment]
@HLeithner 
Do you want me to re-test this?

**Comment by HLeithner** [comment]
> @HLeithner Do you want me to re-test this?

Plz

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on 4edaa4c1d0b730e9ef021f55c99eeedef48ea6f2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43335">issues.joomla.org/tracker/joomla-cms/43335</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 4edaa4c1d0b730e9ef021f55c99eeedef48ea6f2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43335">issues.joomla.org/tracker/joomla-cms/43335</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43335">issues.joomla.org/tracker/joomla-cms/43335</a>.</sub>

**Comment by MacJoom** [comment]
Thank you!

**Pull request merged**